persistence/sql/persister_registration.go

```diff
@@ -64,3 +70,133 @@ func (p *Persister) DeleteExpiredRegistrationFlows(ctx context.Context, expiresA
 	}
 	return nil
 }
+
+func (p *Persister) CreateRegistrationCode(ctx context.Context, codeParams *code.CreateRegistrationCodeParams) (*code.RegistrationCode, error) {
+	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.CreateRegistrationCode")
+	defer span.End()
+
+	now := time.Now()
+
+	registrationCode := &code.RegistrationCode{
+		Address:     codeParams.Address,
+		AddressType: codeParams.AddressType,
+		CodeHMAC:    p.hmacValue(ctx, codeParams.RawCode),
+		IssuedAt:    now,
+		ExpiresAt:   now.UTC().Add(p.r.Config().SelfServiceCodeMethodLifespan(ctx)),
+		FlowID:      codeParams.FlowID,
+		NID:         p.NetworkID(ctx),
+		ID:          uuid.Nil,
+	}
+
+	if err := p.GetConnection(ctx).Create(registrationCode); err != nil {
+		return nil, sqlcon.HandleError(err)
+	}
+	return registrationCode, nil
+}
+
+func (p *Persister) UseRegistrationCode(ctx context.Context, flowID uuid.UUID, rawCode string, addresses ...string) (*code.RegistrationCode, error) {
```

Comment:
This function is very similar to `UseLoginCode`. Can we extract this out into a generic function?

Reply:
I did this already on review branch